Sources/AsyncDNSResolver/c-ares/DNSResolver_c-ares.swift

```diff
@@ -195,6 +195,12 @@ extension Ares {
         private let channel: AresChannel
         private let pollIntervalNanos: UInt64
 
+        private var pollingTask: Task<Void, Error>?
+
+        deinit {
+            self.pollingTask?.cancel()
```

Comment:
This is pretty nasty and shows two issues:

1. This class should have a properly enforced lifecycle but doesn't (which usually also shows up when `weak` suddenly starts becoming required)
2. This class is using `async` but violates Structured Concurrency

Ideally this would implement Structured Concurrency, then cancelling the poll will be automatic (because cancellation propagates down the task tree automatically). And the lifecycle is also taken care of because it starts with the beginning of the scope and ends with the end of the scope.